### PR TITLE
Refactor Layout

### LINKED ISSUE #902: Help wanted, conversion to TS
Hi friends! As of this issue the repo is now ready and supporting migrating page and component files to use TypeScript. For the most part, this will involve:

1. Renaming a file to us a `ts` or `tsx` extension (depending on the presence of JSX)

2. Updating any imports if necessary

3. Removing use of prop types in favor of function component argument defaults

4. Adding a type signature to styled components that consume props, e.g.

  ```tsx
  styled.div<{ $color: string }>`
    color: ${p => p.$color || 'red'};
  `
  ```

If you'd like to claim a file or folder, please comment on this thread to let other people know and start a PR! Thank you very much for helping out 😄

### PR DESCRIPTION
**Pull request by redaktorscha**
Fix Layout component (JS ->TS convertion), 'media' and 'rem' utils; add typings 
fixes #902

### CODE REVIEW

File: utils/media.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { css } from 'styled-components';
+import { RuleSet } from 'styled-components/dist/types';
+
+export const mobile = (inner: ReturnType<typeof css>): RuleSet<object> => css`
```
Comment (probablyup) [inline utils/media.ts:4]: You should be able to remove `RuleSet<object>` for these and use the same `ReturnType<typeof css>` annotation. Perhaps store it to a local type so you can reuse it?

```ts
type CSSUtilType = ReturnType<typeof css>
```

**Comment by probablyup** [comment]
Thanks for helping with this work! Should be good to merge after that last small refactoring.

**Comment by redaktorscha** [comment]
> Thanks for helping with this work! Should be good to merge after that last small refactoring.

fixed & pushed

**Review by probablyup — APPROVED** [review]
Thank you!

**Comment by redaktorscha** [comment]
> Thank you!

you're very welcome! )

**Pull request merged**